src/nvim/path.c

```diff
@@ -1749,12 +1749,22 @@ char_u *find_file_name_in_path(char_u *ptr, size_t len, int options, long count,
   return file_name;
 }
 
-// Check if the "://" of a URL is at the pointer, return URL_SLASH.
+bool path_starts_with_windows_drive_letter(const char *p) {
+    return strlen(p)>=2
+        && ((p[0] >= 'a' && p[0] <= 'z') || (p[0] >= 'A' && p[0] <= 'Z'))
+        && ( p[1] == ':'  || p[1] == '|')
+        && (strlen(p) == 2 || (p[2] == '/' | p[2] == '\\'  | p[2] == '?' |  p[2] == '#'));
```

Comment:
nit: good practice to avoid the redundant strlen() call, though I would expect compiler to figure it out here.

Reply:
can you explain the "?" and "#"? Is `c:?` a valid Windows drive letter path?

Reply:
I'm just keeping the coed close to the rules here https://url.spec.whatwg.org/#start-with-a-windows-drive-letter

also yes ? # are valid according to those rules